src/quantum_siren/pipelines/training/models.py

```diff
@@ -4,23 +4,43 @@
 
 from .ansaetze import ansaetze
 
-import mlflow
+import logging
+
+log = logging.getLogger(__name__)
 
 
 class Model(torch.nn.Module):
     # class Module(torch.nn.Module):
     def __init__(
-        self, n_qubits, shots, vqc_ansatz, iec_ansatz, n_layers, data_reupload
+        self,
+        n_qubits,
+        shots,
+        vqc_ansatz,
+        iec_ansatz,
+        n_layers,
+        data_reupload,
+        output_interpretation,
     ) -> None:
         super().__init__()
 
+        log.info(f"Creating Model with {n_qubits} Qubits, {n_layers} Layers.")
+
         self.shots = None if shots == "None" else shots
         self.n_qubits = n_qubits
         self.n_layers = n_layers
 
         self.iec = getattr(ansaetze, iec_ansatz, ansaetze.nothing)
         self.vqc = getattr(ansaetze, vqc_ansatz, ansaetze.nothing)
 
+        if output_interpretation != "all":
```

Comment:
This check seems very hard-coded. In case you will add another string-type option for `output_interpretation` you will have to remember adapting this check. You might consider trying to cast to int and catching `ValueError` instead.